tensorflow/core/kernels/data/map_dataset_op.cc

```diff
@@ -88,7 +91,9 @@ class MapDatasetOp : public UnaryDatasetOpKernel {
       return output_shapes_;
     }
 
-    string DebugString() const override { return "MapDatasetOp::Dataset"; }
+    string DebugString() const override {
+      return DatasetDebugString(MAP_DATASET);
```

Comment:
if we add `dataset_type_` member for each dataset, will there be some overlap between `dataset_type_` and [type_string_](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/dataset.h#L713)?

Reply:
As far as I can tell, `type_string_` is not used for anything (other than in your unit tests), so it might be okay to get rid of it. Let me try creating a change that does that.

Reply:
Feel free to let me know if there is anything I can help.